manifests/package.pp

```diff
@@ -22,17 +26,41 @@
     }
 
     'RedHat': {
+      $pkg_title = 'sensu'
+      $pkg_name = 'sensu'
+      $pkg_source = undef
+
       class { '::sensu::repo::yum': }
 
       $pkg_require = undef
     }
 
+    'windows': {
+      $repo_require = undef
+
+      $pkg_version = inline_template("<%= scope.lookupvar('sensu::version').sub(/(.*)\\./, '\\1-') %>")
+      $pkg_title = 'sensu'
+      $pkg_name = 'Sensu'
+      $pkg_source = "C:\\Windows\\Temp\\sensu-${pkg_version}.msi"
+      $pkg_require = "Archive[${pkg_source}]"
+
+      archive { $pkg_source:
```

Comment:
In theory yes - the module says it is supported, but in reality would require the users to already have gnu wget for windows installed on their system.

Another option you might want consider would be https://github.com/lwf/puppet-remote_file which is pure ruby so should work on all platforms.